pkg/server/registry/apigroups/acorn/jobs/restart.go

```diff
@@ -0,0 +1,118 @@
+package jobs
+
+import (
+	"context"
+
+	"github.com/acorn-io/mink/pkg/stores"
+	"github.com/acorn-io/mink/pkg/types"
+	apiv1 "github.com/acorn-io/runtime/pkg/apis/api.acorn.io/v1"
+	kclient "github.com/acorn-io/runtime/pkg/k8sclient"
+	"github.com/acorn-io/runtime/pkg/labels"
+	"github.com/acorn-io/z"
+	batchv1 "k8s.io/api/batch/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NewRestart(c client.WithWatch) rest.Storage {
+	return stores.NewBuilder(c.Scheme(), &apiv1.JobRestart{}).
+		WithCreate(&restartStrategy{client: c}).
+		Build()
+}
+
+type restartStrategy struct {
+	client client.WithWatch
+}
+
+func (s *restartStrategy) New() types.Object {
+	return &apiv1.JobRestart{}
+}
+
+func (s *restartStrategy) Create(ctx context.Context, obj types.Object) (types.Object, error) {
+	ri, _ := request.RequestInfoFrom(ctx)
+
+	if ri.Namespace == "" {
+		return obj, nil
+	}
+
+	// Find the job to restart by looking it up with the public name. The only reason we don't set the actual name
+	// to be the public name is because the public name includes a "." which is not allowed in k8s names.
+	job := &apiv1.Job{}
+	err := s.client.Get(ctx, kclient.ObjectKey{Namespace: ri.Namespace, Name: obj.GetLabels()[labels.AcornPublicName]}, job)
+	if err != nil {
+		return nil, err
+	}
+
+	key := kclient.ObjectKey{Namespace: job.Status.JobNamespace, Name: job.Status.JobName}
+
+	cronToRestart := &batchv1.CronJob{}
+	if err = s.client.Get(ctx, key, cronToRestart); err == nil {
+		return obj, s.restartCronJob(ctx, cronToRestart)
+	}
+
+	if !apierrors.IsNotFound(err) {
+		return nil, err
```

Comment:
Do you want to return the `obj` if `err == nil`? Or just always?

Reply:
You are right, we should return the object always.